2017-05-14 17:28:29.047 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:28:29.047 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 17:28:29.606 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:28:29.606 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 17:28:29.606 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 17:28:29.606 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 17:28:29.659 [info] <0.142.0>@group_sup:init:69 ##############
2017-05-14 17:28:29.691 [info] <0.142.0>@group_sup:init:71 ##############
2017-05-14 17:28:29.691 [error] <0.142.0> CRASH REPORT Process <0.142.0> with 0 neighbours exited with reason: no function clause matching lists:foreach(#Fun<group_sup.4.17263849>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in gen_server:init_it/6 line 352
2017-05-14 17:28:29.692 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching lists:foreach(#Fun<group_sup.4.17263849>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in context start_error
2017-05-14 17:28:29.692 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{lists,foreach,[#Fun<group_sup.4.17263849>,{atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,allowed}}}}],[{file,"lists.erl"},{line,1337}]},{group_sup,init,1,[{file,"src/logic/group_sup.erl"},{line,72}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,294}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}} in context start_error
2017-05-14 17:28:29.692 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,group_sup,{function_clause,[{lists,foreach,[#Fun<group_sup.4.17263849>,{atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,allowed}}}}],[{file,"lists.erl"},{line,1337}]},{group_sup,init,1,[{file,"src/logic/group_sup.erl"},{line,72}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,294}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,...}]}}}}},...} in application_master:init/4 line 134
2017-05-14 17:41:41.610 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:41:41.610 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 17:41:42.193 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:41:42.193 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 17:41:42.193 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 17:41:42.193 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 17:41:42.228 [info] <0.142.0>@group_sup:init:69 ##############
2017-05-14 17:41:42.260 [info] <0.142.0>@group_sup:init:71 ##############
2017-05-14 17:41:42.260 [error] <0.142.0> CRASH REPORT Process <0.142.0> with 0 neighbours exited with reason: no function clause matching lists:foreach(#Fun<group_sup.4.17263849>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in gen_server:init_it/6 line 352
2017-05-14 17:41:42.260 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching lists:foreach(#Fun<group_sup.4.17263849>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in context start_error
2017-05-14 17:41:42.261 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{lists,foreach,[#Fun<group_sup.4.17263849>,{atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,allowed}}}}],[{file,"lists.erl"},{line,1337}]},{group_sup,init,1,[{file,"src/logic/group_sup.erl"},{line,72}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,294}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}} in context start_error
2017-05-14 17:43:00.678 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:43:00.678 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 17:43:01.159 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:43:01.159 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 17:43:01.159 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 17:43:01.159 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 17:43:01.163 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 17:43:01.208 [info] <0.143.0>@group_sup:init:69 ##############
2017-05-14 17:43:01.240 [info] <0.143.0>@group_sup:init:71 ##############
2017-05-14 17:43:01.240 [error] <0.143.0> CRASH REPORT Process <0.143.0> with 0 neighbours exited with reason: no function clause matching lists:foreach(#Fun<group_sup.4.17263849>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in gen_server:init_it/6 line 352
2017-05-14 17:43:01.240 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching lists:foreach(#Fun<group_sup.4.17263849>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in context start_error
2017-05-14 17:43:01.241 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{lists,foreach,[#Fun<group_sup.4.17263849>,{atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,allowed}}}}],[{file,"lists.erl"},{line,1337}]},{group_sup,init,1,[{file,"src/logic/group_sup.erl"},{line,72}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,294}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}} in context start_error
2017-05-14 17:46:27.567 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:46:27.567 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 17:46:28.303 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:46:28.303 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 17:46:28.303 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 17:46:28.303 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 17:46:28.306 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 17:46:28.324 [info] <0.142.0>@group_sup:init:69 ##############
2017-05-14 17:46:28.369 [info] <0.142.0>@group_sup:init:71 ################
2017-05-14 17:46:28.369 [info] <0.142.0>@group_sup:init:72 ~p ~n
2017-05-14 17:46:28.369 [error] <0.142.0> CRASH REPORT Process <0.142.0> with 0 neighbours exited with reason: no function clause matching lists:foreach(#Fun<group_sup.6.114317290>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in gen_server:init_it/6 line 352
2017-05-14 17:46:28.369 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching lists:foreach(#Fun<group_sup.6.114317290>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in context start_error
2017-05-14 17:46:28.369 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{lists,foreach,[#Fun<group_sup.6.114317290>,{atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,allowed}}}}],[{file,"lists.erl"},{line,1337}]},{group_sup,init,1,[{file,"src/logic/group_sup.erl"},{line,73}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,294}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}} in context start_error
2017-05-14 17:46:28.370 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,group_sup,{function_clause,[{lists,foreach,[#Fun<group_sup.6.114317290>,{atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,allowed}}}}],[{file,"lists.erl"},{line,1337}]},{group_sup,init,1,[{file,"src/logic/group_sup.erl"},{line,73}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,294}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,...}]}}}}},...} in application_master:init/4 line 134
2017-05-14 17:46:48.400 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:46:48.400 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 17:46:49.234 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 17:46:49.234 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 17:46:49.234 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 17:46:49.235 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 17:46:49.237 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 17:46:49.259 [info] <0.142.0>@group_sup:init:69 ##############
2017-05-14 17:46:49.294 [info] <0.142.0>@group_sup:init:71 ################
2017-05-14 17:46:49.294 [info] <0.142.0>@group_sup:init:72 {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,allowed}}}} 
2017-05-14 17:46:49.294 [error] <0.142.0> CRASH REPORT Process <0.142.0> with 0 neighbours exited with reason: no function clause matching lists:foreach(#Fun<group_sup.6.39345746>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in gen_server:init_it/6 line 352
2017-05-14 17:46:49.294 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching lists:foreach(#Fun<group_sup.6.39345746>, {atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,...}}}}) line 1337 in context start_error
2017-05-14 17:46:49.294 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{lists,foreach,[#Fun<group_sup.6.39345746>,{atomic,{qlc_handle,{qlc_lc,#Fun<mod_group_sup.1.124098997>,{qlc_opt,false,false,-1,any,[],any,524288,allowed}}}}],[{file,"lists.erl"},{line,1337}]},{group_sup,init,1,[{file,"src/logic/group_sup.erl"},{line,73}]},{supervisor,init,1,[{file,"supervisor.erl"},{line,294}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}} in context start_error
2017-05-14 18:48:30.780 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 18:48:30.780 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 18:48:31.232 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 18:48:31.232 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 18:48:31.232 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 18:48:31.232 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 18:48:31.235 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 18:48:31.261 [info] <0.145.0>@group_sup:init:69 ##############
2017-05-14 18:48:31.266 [info] <0.145.0>@group_sup:init:71 ################
2017-05-14 18:48:31.266 [info] <0.145.0>@group_sup:init:72 [] 
2017-05-14 18:48:31.267 [error] <0.145.0> CRASH REPORT Process <0.145.0> with 0 neighbours exited with reason: no function clause matching supervisor:check_startspec(ok, []) line 1295 in gen_server:init_it/6 line 352
2017-05-14 18:48:31.267 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching supervisor:check_startspec(ok, []) line 1295 in context start_error
2017-05-14 18:48:31.267 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{supervisor,check_startspec,[ok,[]],[{file,"supervisor.erl"},{line,1295}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,312}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}} in context start_error
2017-05-14 18:51:18.163 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 18:51:18.163 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 18:51:18.950 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 18:51:18.950 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 18:51:18.950 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 18:51:18.950 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 18:51:18.953 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 18:51:18.965 [info] <0.145.0>@group_sup:init:69 ##############
2017-05-14 18:51:18.988 [info] <0.145.0>@group_sup:init:71 ################
2017-05-14 18:51:18.988 [info] <0.145.0>@group_sup:init:72 [] 
2017-05-14 18:51:18.988 [info] <0.145.0>@group_sup:init:79 ok 
2017-05-14 18:51:18.988 [error] <0.145.0> CRASH REPORT Process <0.145.0> with 0 neighbours exited with reason: no function clause matching supervisor:check_startspec(ok, []) line 1295 in gen_server:init_it/6 line 352
2017-05-14 18:51:18.988 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching supervisor:check_startspec(ok, []) line 1295 in context start_error
2017-05-14 18:51:18.989 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{supervisor,check_startspec,[ok,[]],[{file,"supervisor.erl"},{line,1295}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,312}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}} in context start_error
2017-05-14 18:52:15.339 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 18:52:15.339 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 18:52:15.871 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 18:52:15.871 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 18:52:15.871 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 18:52:15.872 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 18:52:15.874 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 18:52:15.889 [info] <0.145.0>@group_sup:init:69 ##############
2017-05-14 18:52:15.923 [info] <0.145.0>@group_sup:init:71 ################
2017-05-14 18:52:15.923 [info] <0.145.0>@group_sup:init:72 [] 
2017-05-14 18:52:15.923 [info] <0.145.0>@group_sup:init:79 [] 
2017-05-14 18:52:15.924 [error] <0.99.0> Supervisor control_sup had child user_sup started with user_sup:start_link() at undefined exit with reason {'EXIT',{undef,[{user_sup,start_link,[],[]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,365}]},{supervisor,start_children,3,[{file,"supervisor.erl"},{line,348}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,314}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}} in context start_error
2017-05-14 18:54:19.054 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 18:54:19.054 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 18:54:19.507 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 18:54:19.507 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 18:54:19.507 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 18:54:19.508 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 18:54:19.511 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 18:54:19.526 [info] <0.145.0>@group_sup:init:69 ##############
2017-05-14 18:54:19.571 [info] <0.145.0>@group_sup:init:71 ################
2017-05-14 18:54:19.571 [info] <0.145.0>@group_sup:init:72 [] 
2017-05-14 18:54:19.571 [info] <0.145.0>@group_sup:init:80 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-14 18:54:19.589 [error] <0.145.0> Supervisor group_sup had child 1 started with group:start_link(1) at undefined exit with reason {'EXIT',{undef,[{group,start_link,[1],[]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,365}]},{supervisor,start_children,3,[{file,"supervisor.erl"},{line,348}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,314}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}} in context start_error
2017-05-14 18:54:19.590 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,1,{'EXIT',{undef,[{group,start_link,[1],[]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,365}]},{supervisor,start_children,3,[{file,"supervisor.erl"},{line,348}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,314}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}}} in context start_error
2017-05-14 19:00:11.231 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 19:00:11.231 [info] <0.100.0>@chat_sup:init:26 ###############
2017-05-14 19:00:11.753 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 19:00:11.753 [info] <0.101.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 19:00:11.753 [info] <0.101.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 19:00:11.753 [info] <0.101.0>@mod_control_sup:init:38 error
2017-05-14 19:00:11.756 [info] <0.101.0>@control_sup:init:55 ##########
2017-05-14 19:00:11.781 [info] <0.147.0>@group_sup:init:69 ##############
2017-05-14 19:00:11.804 [info] <0.147.0>@group_sup:init:71 ################
2017-05-14 19:00:11.804 [info] <0.147.0>@group_sup:init:72 [] 
2017-05-14 19:00:11.804 [info] <0.147.0>@group_sup:init:80 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-14 19:00:11.805 [error] <0.147.0> Supervisor group_sup had child 2 started with group:start_link(2) at undefined exit with reason {already_started,<0.149.0>} in context start_error
2017-05-14 19:00:11.805 [error] <0.101.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,2,{already_started,<0.149.0>}}} in context start_error
2017-05-14 19:00:11.805 [error] <0.100.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{shutdown,{failed_to_start_child,2,{already_started,<0.149.0>}}}}} in context start_error
2017-05-14 19:00:11.806 [error] <0.98.0> CRASH REPORT Process <0.98.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,group_sup,{shutdown,{failed_to_start_child,2,{already_started,<0.149.0>}}}}}}},{chat_app,start,[normal,[]]}} in application_master:init/4 line 134
2017-05-14 19:00:11.806 [info] <0.31.0> Application chat exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,group_sup,{shutdown,{failed_to_start_child,2,{already_started,<0.149.0>}}}}}}},{chat_app,start,[normal,[]]}}
2017-05-14 19:01:03.433 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 19:01:03.433 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 19:01:03.865 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 19:01:03.865 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 19:01:03.865 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 19:01:03.866 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 19:01:03.868 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 19:01:03.893 [info] <0.145.0>@group_sup:init:69 ##############
2017-05-14 19:01:03.903 [info] <0.145.0>@group_sup:init:71 ################
2017-05-14 19:01:03.903 [info] <0.145.0>@group_sup:init:72 [] 
2017-05-14 19:01:03.903 [info] <0.145.0>@group_sup:init:80 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-14 19:01:03.904 [error] <0.99.0> Supervisor control_sup had child user_sup started with user_sup:start_link() at undefined exit with reason {'EXIT',{undef,[{user_sup,start_link,[],[]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,365}]},{supervisor,start_children,3,[{file,"supervisor.erl"},{line,348}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,314}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}} in context start_error
2017-05-14 19:01:03.905 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,user_sup,{'EXIT',{undef,[{user_sup,start_link,[],[]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,365}]},{supervisor,start_children,3,[{file,"supervisor.erl"},{line,348}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,314}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}}} in context start_error
2017-05-14 19:01:03.905 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,user_sup,{'EXIT',{undef,[{user_sup,start_link,[],[]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,365}]},{supervisor,start_children,3,[{file,"supervisor.erl"},{line,348}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,314}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}}}}},{chat_app,start,[normal,[]]}} in application_master:init/4 line 134
2017-05-14 19:13:39.473 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 19:13:39.473 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 19:13:40.032 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 19:13:40.032 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 19:13:40.032 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 19:13:40.033 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 19:13:40.041 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 19:13:40.043 [info] <0.145.0>@group_sup:init:69 ##############
2017-05-14 19:13:40.047 [info] <0.145.0>@group_sup:init:71 ################
2017-05-14 19:13:40.047 [info] <0.145.0>@group_sup:init:72 [] 
2017-05-14 19:13:40.047 [info] <0.145.0>@group_sup:init:80 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-14 19:13:40.082 [error] <0.150.0> Supervisor client_sup had child 'AName' started with 'AModule':start_link() at undefined exit with reason {'EXIT',{undef,[{'AModule',start_link,[],[]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,365}]},{supervisor,start_children,3,[{file,"supervisor.erl"},{line,348}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,314}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}} in context start_error
2017-05-14 19:13:40.083 [error] <0.99.0> Supervisor control_sup had child client_sup started with client_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,'AName',{'EXIT',{undef,[{'AModule',start_link,[],[]},{supervisor,do_start_child,2,[{file,"supervisor.erl"},{line,365}]},{supervisor,start_children,3,[{file,"supervisor.erl"},{line,348}]},{supervisor,init_children,2,[{file,"supervisor.erl"},{line,314}]},{gen_server,init_it,6,[{file,"gen_server.erl"},{line,328}]},{proc_lib,init_p_do_apply,3,[{file,"proc_lib.erl"},{line,247}]}]}}}} in context start_error
2017-05-14 19:18:49.306 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-14 19:18:49.306 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-14 19:18:50.009 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-14 19:18:50.009 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-14 19:18:50.010 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-14 19:18:50.010 [info] <0.99.0>@mod_control_sup:init:38 error
2017-05-14 19:18:50.013 [info] <0.99.0>@control_sup:init:55 ##########
2017-05-14 19:18:50.024 [info] <0.145.0>@group_sup:init:69 ##############
2017-05-14 19:18:50.094 [info] <0.145.0>@group_sup:init:71 ################
2017-05-14 19:18:50.094 [info] <0.145.0>@group_sup:init:72 [] 
2017-05-14 19:18:50.094 [info] <0.145.0>@group_sup:init:80 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-14 19:18:50.112 [error] <0.150.0> CRASH REPORT Process <0.150.0> with 0 neighbours exited with reason: {start_spec,{invalid_mfa,{client,[]}}} in gen_server:init_it/6 line 344
2017-05-14 19:18:50.112 [error] <0.99.0> Supervisor control_sup had child client_sup started with client_sup:start_link() at undefined exit with reason {start_spec,{invalid_mfa,{client,[]}}} in context start_error
2017-05-14 19:18:50.112 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,client_sup,{start_spec,{invalid_mfa,{client,[]}}}}} in context start_error
2017-05-15 10:48:04.496 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-15 10:48:04.496 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-15 10:48:05.334 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-15 10:48:05.334 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-15 10:48:05.334 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-15 10:48:05.632 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-15 10:48:05.632 [info] <0.99.0>@control_sup:init:55 
2017-05-15 10:48:05.765 [info] <0.147.0>@group_sup:init:70 ################
2017-05-15 10:48:05.765 [info] <0.147.0>@group_sup:init:71 [] 
2017-05-15 10:48:05.765 [info] <0.147.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:05.802 [error] <0.149.0> gen_server <0.149.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:05.802 [error] <0.150.0> gen_server <0.150.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:05.802 [error] <0.151.0> gen_server <0.151.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:05.802 [info] <0.31.0> Application chat started on node 'yangwei@brucedeiMac.local'
2017-05-15 10:48:06.135 [error] <0.149.0> CRASH REPORT Process <0.149.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.135 [error] <0.151.0> CRASH REPORT Process <0.151.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.135 [error] <0.150.0> CRASH REPORT Process <0.150.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.135 [error] <0.147.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.149.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.135 [error] <0.153.0> gen_server <0.153.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:06.135 [error] <0.153.0> CRASH REPORT Process <0.153.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.135 [error] <0.147.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.151.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.135 [error] <0.154.0> gen_server <0.154.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:06.135 [error] <0.154.0> CRASH REPORT Process <0.154.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.135 [error] <0.147.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.150.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.136 [error] <0.155.0> gen_server <0.155.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:06.136 [error] <0.155.0> CRASH REPORT Process <0.155.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.136 [error] <0.147.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.153.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.136 [error] <0.156.0> gen_server <0.156.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:06.136 [error] <0.156.0> CRASH REPORT Process <0.156.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.136 [error] <0.147.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.154.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.136 [error] <0.157.0> gen_server <0.157.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:06.136 [error] <0.157.0> CRASH REPORT Process <0.157.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.136 [error] <0.147.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.155.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.136 [error] <0.158.0> gen_server <0.158.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:06.136 [error] <0.158.0> CRASH REPORT Process <0.158.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.136 [error] <0.147.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.156.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.136 [error] <0.159.0> gen_server <0.159.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:06.136 [error] <0.159.0> CRASH REPORT Process <0.159.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.136 [error] <0.147.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.157.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.137 [error] <0.160.0> gen_server <0.160.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:06.137 [error] <0.160.0> CRASH REPORT Process <0.160.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.137 [error] <0.147.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.158.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.137 [error] <0.161.0> gen_server <0.161.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:06.137 [error] <0.161.0> CRASH REPORT Process <0.161.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.137 [error] <0.147.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.159.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.137 [error] <0.162.0> gen_server <0.162.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:06.137 [error] <0.162.0> CRASH REPORT Process <0.162.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.137 [error] <0.147.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.160.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.137 [error] <0.163.0> gen_server <0.163.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:06.137 [error] <0.163.0> CRASH REPORT Process <0.163.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.137 [error] <0.147.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.161.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.137 [error] <0.164.0> gen_server <0.164.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:06.137 [error] <0.164.0> CRASH REPORT Process <0.164.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.137 [error] <0.147.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.162.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.137 [error] <0.165.0> gen_server <0.165.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:06.138 [error] <0.165.0> CRASH REPORT Process <0.165.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.138 [error] <0.147.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.163.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.138 [error] <0.166.0> gen_server <0.166.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:06.138 [error] <0.166.0> CRASH REPORT Process <0.166.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.138 [error] <0.147.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.164.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.138 [error] <0.167.0> gen_server <0.167.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:06.138 [error] <0.167.0> CRASH REPORT Process <0.167.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.138 [error] <0.147.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.165.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.138 [error] <0.168.0> gen_server <0.168.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:06.138 [error] <0.168.0> CRASH REPORT Process <0.168.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:06.138 [error] <0.147.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.166.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:06.180 [info] <0.1153.0>@group_sup:init:70 ################
2017-05-15 10:48:06.180 [info] <0.1153.0>@group_sup:init:71 [] 
2017-05-15 10:48:06.180 [info] <0.1153.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:06.224 [info] <0.2158.0>@group_sup:init:70 ################
2017-05-15 10:48:06.224 [info] <0.2158.0>@group_sup:init:71 [] 
2017-05-15 10:48:06.224 [info] <0.2158.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:06.265 [info] <0.3163.0>@group_sup:init:70 ################
2017-05-15 10:48:06.265 [info] <0.3163.0>@group_sup:init:71 [] 
2017-05-15 10:48:06.265 [info] <0.3163.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:06.305 [info] <0.4168.0>@group_sup:init:70 ################
2017-05-15 10:48:06.305 [info] <0.4168.0>@group_sup:init:71 [] 
2017-05-15 10:48:06.305 [info] <0.4168.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:06.349 [info] <0.5173.0>@group_sup:init:70 ################
2017-05-15 10:48:06.349 [info] <0.5173.0>@group_sup:init:71 [] 
2017-05-15 10:48:06.349 [info] <0.5173.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:06.388 [info] <0.6178.0>@group_sup:init:70 ################
2017-05-15 10:48:19.859 [info] <0.6178.0>@group_sup:init:71 [] 
2017-05-15 10:48:19.859 [info] <0.6178.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:19.859 [warning] <0.30.0> lager_error_logger_h dropped 14038 messages in the last second that exceeded the limit of 50 messages/sec
2017-05-15 10:48:19.859 [error] <0.6180.0> gen_server <0.6180.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:19.859 [error] <0.6180.0> CRASH REPORT Process <0.6180.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.859 [error] <0.6181.0> gen_server <0.6181.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:19.860 [error] <0.6181.0> CRASH REPORT Process <0.6181.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.860 [error] <0.6182.0> gen_server <0.6182.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:19.860 [error] <0.6182.0> CRASH REPORT Process <0.6182.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.860 [error] <0.6178.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.6180.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.860 [error] <0.6183.0> gen_server <0.6183.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:19.860 [error] <0.6183.0> CRASH REPORT Process <0.6183.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.860 [error] <0.6178.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.6181.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.860 [error] <0.6184.0> gen_server <0.6184.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:19.860 [error] <0.6184.0> CRASH REPORT Process <0.6184.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.860 [error] <0.6178.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.6182.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.861 [error] <0.6185.0> gen_server <0.6185.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:19.861 [error] <0.6185.0> CRASH REPORT Process <0.6185.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.861 [error] <0.6178.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.6183.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.861 [error] <0.6186.0> gen_server <0.6186.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:19.861 [error] <0.6186.0> CRASH REPORT Process <0.6186.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.861 [error] <0.6178.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.6184.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.861 [error] <0.6187.0> gen_server <0.6187.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:19.861 [error] <0.6187.0> CRASH REPORT Process <0.6187.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.861 [error] <0.6178.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.6185.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.861 [error] <0.6188.0> gen_server <0.6188.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:19.862 [error] <0.6188.0> CRASH REPORT Process <0.6188.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.862 [error] <0.6178.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.6186.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.862 [error] <0.6189.0> gen_server <0.6189.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:19.862 [error] <0.6189.0> CRASH REPORT Process <0.6189.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.862 [error] <0.6178.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.6187.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.862 [error] <0.6190.0> gen_server <0.6190.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:19.862 [error] <0.6190.0> CRASH REPORT Process <0.6190.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.862 [error] <0.6178.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.6188.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.862 [error] <0.6191.0> gen_server <0.6191.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:19.862 [error] <0.6191.0> CRASH REPORT Process <0.6191.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.862 [error] <0.6178.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.6189.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:19.862 [error] <0.6192.0> gen_server <0.6192.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:19.862 [error] <0.6192.0> CRASH REPORT Process <0.6192.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:19.907 [info] <0.7183.0>@group_sup:init:70 ################
2017-05-15 10:48:19.907 [info] <0.7183.0>@group_sup:init:71 [] 
2017-05-15 10:48:19.907 [info] <0.7183.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:19.953 [info] <0.8188.0>@group_sup:init:70 ################
2017-05-15 10:48:19.953 [info] <0.8188.0>@group_sup:init:71 [] 
2017-05-15 10:48:19.953 [info] <0.8188.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:19.998 [info] <0.9193.0>@group_sup:init:70 ################
2017-05-15 10:48:19.998 [info] <0.9193.0>@group_sup:init:71 [] 
2017-05-15 10:48:19.998 [info] <0.9193.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:20.000 [warning] <0.30.0> lager_error_logger_h dropped 7461 messages in the last second that exceeded the limit of 50 messages/sec
2017-05-15 10:48:20.000 [error] <0.9262.0> gen_server <0.9262.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:20.000 [error] <0.9262.0> CRASH REPORT Process <0.9262.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:20.000 [error] <0.9193.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.9260.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:20.000 [error] <0.9263.0> gen_server <0.9263.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:20.000 [error] <0.9263.0> CRASH REPORT Process <0.9263.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:20.000 [error] <0.9193.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.9261.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:20.000 [error] <0.9264.0> gen_server <0.9264.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:20.000 [error] <0.9264.0> CRASH REPORT Process <0.9264.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:20.000 [error] <0.9193.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.9262.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:20.000 [error] <0.9265.0> gen_server <0.9265.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:20.001 [error] <0.9265.0> CRASH REPORT Process <0.9265.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:20.001 [error] <0.9193.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.9263.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:20.001 [error] <0.9266.0> gen_server <0.9266.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:20.001 [error] <0.9266.0> CRASH REPORT Process <0.9266.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:20.001 [error] <0.9193.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.9264.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:20.001 [error] <0.9267.0> gen_server <0.9267.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:20.001 [error] <0.9267.0> CRASH REPORT Process <0.9267.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:20.001 [error] <0.9193.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.9265.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:20.001 [error] <0.9268.0> gen_server <0.9268.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:20.001 [error] <0.9268.0> CRASH REPORT Process <0.9268.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:20.001 [error] <0.9193.0> Supervisor group_sup had child 3 started with group:start_link(3) at <0.9266.0> exit with reason no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in context child_terminated
2017-05-15 10:48:20.001 [error] <0.9269.0> gen_server <0.9269.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101
2017-05-15 10:48:20.001 [error] <0.9269.0> CRASH REPORT Process <0.9269.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,3,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:20.002 [error] <0.9193.0> Supervisor group_sup had child 1 started with group:start_link(1) at <0.9267.0> exit with reason no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in context child_terminated
2017-05-15 10:48:20.002 [error] <0.9270.0> gen_server <0.9270.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101
2017-05-15 10:48:20.002 [error] <0.9270.0> CRASH REPORT Process <0.9270.0> with 0 neighbours exited with reason: no function clause matching group:handle_cast({online}, {state,1,undefined}) line 101 in gen_server:terminate/7 line 812
2017-05-15 10:48:20.002 [error] <0.9193.0> Supervisor group_sup had child 2 started with group:start_link(2) at <0.9268.0> exit with reason no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101 in context child_terminated
2017-05-15 10:48:20.002 [error] <0.9271.0> gen_server <0.9271.0> terminated with reason: no function clause matching group:handle_cast({online}, {state,2,undefined}) line 101
2017-05-15 10:48:39.485 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-15 10:48:39.485 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-15 10:48:40.145 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-15 10:48:40.145 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-15 10:48:40.145 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-15 10:48:40.145 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-15 10:48:40.148 [info] <0.99.0>@control_sup:init:55 
2017-05-15 10:48:40.255 [info] <0.149.0>@group_sup:init:70 ################
2017-05-15 10:48:40.255 [info] <0.149.0>@group_sup:init:71 [] 
2017-05-15 10:48:40.255 [info] <0.149.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:48:40.256 [info] <0.31.0> Application chat started on node 'yangwei@brucedeiMac.local'
2017-05-15 10:49:16.964 [info] <0.31.0> Application lager started on node 'yangwei@brucedeiMac.local'
2017-05-15 10:49:16.964 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-15 10:49:17.453 [info] <0.31.0> Application mnesia started on node 'yangwei@brucedeiMac.local'
2017-05-15 10:49:17.454 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei@brucedeiMac.local'
2017-05-15 10:49:17.454 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei@brucedeiMac.local',{already_exists,'yangwei@brucedeiMac.local'}}}
2017-05-15 10:49:17.454 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-15 10:49:17.456 [info] <0.99.0>@control_sup:init:55 
2017-05-15 10:49:17.511 [info] <0.146.0>@group_sup:init:70 ################
2017-05-15 10:49:17.511 [info] <0.146.0>@group_sup:init:71 [] 
2017-05-15 10:49:17.511 [info] <0.146.0>@group_sup:init:79 [{1,{group,start_link,[1]},permanent,2000,worker,[group]},{2,{group,start_link,[2]},permanent,2000,worker,[group]},{3,{group,start_link,[3]},permanent,2000,worker,[group]}] 
2017-05-15 10:49:17.512 [info] <0.148.0>@group:handle_cast:104 <0.148.0>
2017-05-15 10:49:17.512 [info] <0.149.0>@group:handle_cast:104 <0.149.0>
2017-05-15 10:49:17.512 [info] <0.150.0>@group:handle_cast:104 <0.150.0>
2017-05-15 10:49:17.512 [info] <0.31.0> Application chat started on node 'yangwei@brucedeiMac.local'
2017-05-15 11:48:53.962 [info] <0.31.0> Application chat exited with reason: stopped
2017-05-16 18:12:47.780 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-16 18:12:47.780 [info] <0.98.0>@chat_sup:init:26 ###############
2017-05-16 18:12:47.965 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-16 18:12:47.965 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei_node_01@brucedeiMac.local'
2017-05-16 18:12:47.965 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei_node_01@brucedeiMac.local',{already_exists,'yangwei_node_01@brucedeiMac.local'}}}
2017-05-16 18:12:47.972 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-16 18:12:47.973 [info] <0.99.0>@control_sup:init:58 
2017-05-16 18:12:48.003 [info] <0.120.0>@group_sup:init:70 [] 
2017-05-16 18:12:48.003 [info] <0.120.0>@group_sup:init:78 [] 
2017-05-16 18:12:48.049 [info] <0.31.0> Application chat started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 20:51:16.973 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 20:51:16.973 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 20:51:17.058 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 20:51:17.058 [info] <0.99.0>@mod_control_sup:init:25 node: 'yangwei_node_01@brucedeiMac.local'
2017-05-17 20:51:17.058 [info] <0.99.0>@mod_control_sup:init:26 reply: {error,{'yangwei_node_01@brucedeiMac.local',{already_exists,'yangwei_node_01@brucedeiMac.local'}}}
2017-05-17 20:51:17.065 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 20:51:17.065 [info] <0.99.0>@control_sup:init:61 
2017-05-17 20:51:17.119 [error] <0.120.0> CRASH REPORT Process <0.120.0> with 0 neighbours exited with reason: no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in gen_server:init_it/6 line 352
2017-05-17 20:51:17.119 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in context start_error
2017-05-17 20:51:17.119 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{mod_mnesia,'-get_group_keys/0-lc$^0/1-0-',[{qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.64458334>,#Fun<mnesia.25.64458334>,#Fun<mnesia.28.64458334>,#Fun<mnesia.27.64458334>,#Fun<mnesia.26.64458334>,'=:=',undefined,no_match_spec}}],[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{mod_mnesia,get_group_keys,0,[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{group_sup,init,1,[{file,"src/logic..."},...]},...]}}} in context start_error
2017-05-17 20:51:17.119 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,group_sup,{function_clause,[{mod_mnesia,'-get_group_keys/0-lc$^0/1-0-',[{qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.64458334>,#Fun<mnesia.25.64458334>,#Fun<mnesia.28.64458334>,#Fun<mnesia.27.64458334>,#Fun<mnesia.26.64458334>,'=:=',undefined,no_match_spec}}],[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{mod_mnesia,get_group_keys,0,[{file,"src/mod/mod_mnesia.erl"},{...}]},...]}}}}},...} in application_master:init/4 line 134
2017-05-17 20:52:22.630 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 20:52:22.630 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 20:52:22.722 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 20:52:22.729 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 20:52:22.729 [info] <0.99.0>@control_sup:init:61 
2017-05-17 20:52:22.777 [error] <0.120.0> CRASH REPORT Process <0.120.0> with 0 neighbours exited with reason: no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in gen_server:init_it/6 line 352
2017-05-17 20:52:22.777 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in context start_error
2017-05-17 20:52:22.777 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{mod_mnesia,'-get_group_keys/0-lc$^0/1-0-',[{qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.64458334>,#Fun<mnesia.25.64458334>,#Fun<mnesia.28.64458334>,#Fun<mnesia.27.64458334>,#Fun<mnesia.26.64458334>,'=:=',undefined,no_match_spec}}],[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{mod_mnesia,get_group_keys,0,[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{group_sup,init,1,[{file,"src/logic..."},...]},...]}}} in context start_error
2017-05-17 21:03:29.973 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:03:29.973 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:03:30.069 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:03:30.076 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:03:30.076 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:03:30.108 [error] <0.120.0> CRASH REPORT Process <0.120.0> with 0 neighbours exited with reason: no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in gen_server:init_it/6 line 352
2017-05-17 21:03:30.108 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in context start_error
2017-05-17 21:03:30.108 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{mod_mnesia,'-get_group_keys/0-lc$^0/1-0-',[{qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.64458334>,#Fun<mnesia.25.64458334>,#Fun<mnesia.28.64458334>,#Fun<mnesia.27.64458334>,#Fun<mnesia.26.64458334>,'=:=',undefined,no_match_spec}}],[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{mod_mnesia,get_group_keys,0,[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{group_sup,init,1,[{file,"src/logic..."},...]},...]}}} in context start_error
2017-05-17 21:03:47.374 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:03:47.374 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:03:47.472 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:03:47.479 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:03:47.479 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:03:47.508 [error] <0.119.0> CRASH REPORT Process <0.119.0> with 0 neighbours exited with reason: no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in gen_server:init_it/6 line 352
2017-05-17 21:03:47.509 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in context start_error
2017-05-17 21:03:47.509 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{mod_mnesia,'-get_group_keys/0-lc$^0/1-0-',[{qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.64458334>,#Fun<mnesia.25.64458334>,#Fun<mnesia.28.64458334>,#Fun<mnesia.27.64458334>,#Fun<mnesia.26.64458334>,'=:=',undefined,no_match_spec}}],[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{mod_mnesia,get_group_keys,0,[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{group_sup,init,1,[{file,"src/logic..."},...]},...]}}} in context start_error
2017-05-17 21:03:47.509 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,group_sup,{function_clause,[{mod_mnesia,'-get_group_keys/0-lc$^0/1-0-',[{qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.64458334>,#Fun<mnesia.25.64458334>,#Fun<mnesia.28.64458334>,#Fun<mnesia.27.64458334>,#Fun<mnesia.26.64458334>,'=:=',undefined,no_match_spec}}],[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{mod_mnesia,get_group_keys,0,[{file,"src/mod/mod_mnesia.erl"},{...}]},...]}}}}},...} in application_master:init/4 line 134
2017-05-17 21:05:33.523 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:05:33.523 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:05:33.929 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:05:33.938 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:05:33.939 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:05:34.056 [error] <0.119.0> CRASH REPORT Process <0.119.0> with 0 neighbours exited with reason: no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in gen_server:init_it/6 line 352
2017-05-17 21:05:34.056 [error] <0.99.0> Supervisor control_sup had child group_sup started with group_sup:start_link() at undefined exit with reason no function clause matching mod_mnesia:'-get_group_keys/0-lc$^0/1-0-'({qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.6445833..>,...}}) line 17 in context start_error
2017-05-17 21:05:34.057 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,group_sup,{function_clause,[{mod_mnesia,'-get_group_keys/0-lc$^0/1-0-',[{qlc_handle,{qlc_table,#Fun<mnesia.22.64458334>,true,#Fun<mnesia.23.64458334>,#Fun<mnesia.24.64458334>,#Fun<mnesia.25.64458334>,#Fun<mnesia.28.64458334>,#Fun<mnesia.27.64458334>,#Fun<mnesia.26.64458334>,'=:=',undefined,no_match_spec}}],[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{mod_mnesia,get_group_keys,0,[{file,"src/mod/mod_mnesia.erl"},{line,17}]},{group_sup,init,1,[{file,"src/logic..."},...]},...]}}} in context start_error
2017-05-17 21:08:27.460 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:08:27.460 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:08:27.566 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:08:27.573 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:08:27.573 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:08:27.583 [error] <0.120.0> CRASH REPORT Process <0.120.0> with 0 neighbours exited with reason: {bad_start_spec,[]} in gen_server:init_it/6 line 344
2017-05-17 21:08:27.583 [error] <0.99.0> Supervisor control_sup had child client_sup started with client_sup:start_link() at undefined exit with reason {bad_start_spec,[]} in context start_error
2017-05-17 21:08:27.583 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}} in context start_error
2017-05-17 21:08:27.584 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}}}},{chat_app,start,[normal,[]]}} in application_master:init/4 line 134
2017-05-17 21:08:27.584 [info] <0.31.0> Application chat exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}}}},{chat_app,start,[normal,[]]}}
2017-05-17 21:09:42.806 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:09:42.806 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:09:42.901 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:09:42.907 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:09:42.908 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:09:42.908 [info] <0.119.0>@group_sup:init:82 ########
2017-05-17 21:09:42.909 [info] <0.120.0>@client_sup:init:69 ########
2017-05-17 21:09:42.909 [error] <0.120.0> CRASH REPORT Process <0.120.0> with 0 neighbours exited with reason: {bad_start_spec,[]} in gen_server:init_it/6 line 344
2017-05-17 21:09:42.909 [error] <0.99.0> Supervisor control_sup had child client_sup started with client_sup:start_link() at undefined exit with reason {bad_start_spec,[]} in context start_error
2017-05-17 21:09:42.909 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}} in context start_error
2017-05-17 21:09:42.910 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}}}},{chat_app,start,[normal,[]]}} in application_master:init/4 line 134
2017-05-17 21:11:04.520 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:11:04.520 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:11:04.618 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:11:04.624 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:11:04.625 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:11:04.627 [info] <0.120.0>@group_sup:init:82 ######## group_sup
2017-05-17 21:11:04.628 [info] <0.121.0>@client_sup:init:69 ######## client_sup
2017-05-17 21:11:04.628 [error] <0.121.0> CRASH REPORT Process <0.121.0> with 0 neighbours exited with reason: {bad_start_spec,[]} in gen_server:init_it/6 line 344
2017-05-17 21:11:04.628 [error] <0.99.0> Supervisor control_sup had child client_sup started with client_sup:start_link() at undefined exit with reason {bad_start_spec,[]} in context start_error
2017-05-17 21:11:04.628 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}} in context start_error
2017-05-17 21:12:18.433 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:12:18.433 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:12:18.485 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:12:18.491 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:12:18.492 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:12:18.492 [info] <0.119.0>@id_generator:init:30 ########### id_generator
2017-05-17 21:12:18.493 [info] <0.120.0>@group_sup:init:82 ######## group_sup
2017-05-17 21:12:18.494 [info] <0.121.0>@client_sup:init:69 ######## client_sup
2017-05-17 21:12:18.494 [error] <0.121.0> CRASH REPORT Process <0.121.0> with 0 neighbours exited with reason: {bad_start_spec,[]} in gen_server:init_it/6 line 344
2017-05-17 21:12:18.494 [error] <0.99.0> Supervisor control_sup had child client_sup started with client_sup:start_link() at undefined exit with reason {bad_start_spec,[]} in context start_error
2017-05-17 21:12:18.494 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}} in context start_error
2017-05-17 21:12:18.494 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}}}},{chat_app,start,[normal,[]]}} in application_master:init/4 line 134
2017-05-17 21:12:18.494 [info] <0.31.0> Application chat exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}}}},{chat_app,start,[normal,[]]}}
2017-05-17 21:13:26.036 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:13:26.036 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:13:26.090 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:13:26.096 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:13:26.097 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:13:26.098 [info] <0.119.0>@id_generator:init:30 ########### id_generator
2017-05-17 21:13:26.098 [info] <0.120.0>@group_sup:init:82 ######## group_sup
2017-05-17 21:13:26.099 [info] <0.121.0>@client_sup:init:69 ######## client_sup
2017-05-17 21:13:26.099 [error] <0.121.0> CRASH REPORT Process <0.121.0> with 0 neighbours exited with reason: {bad_start_spec,[]} in gen_server:init_it/6 line 344
2017-05-17 21:13:26.099 [error] <0.99.0> Supervisor control_sup had child client_sup started with client_sup:start_link() at undefined exit with reason {bad_start_spec,[]} in context start_error
2017-05-17 21:13:26.099 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}} in context start_error
2017-05-17 21:13:26.099 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}}}},{chat_app,start,[normal,[]]}} in application_master:init/4 line 134
2017-05-17 21:14:33.227 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:14:33.227 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:14:33.295 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:14:33.302 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:14:33.303 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:14:33.310 [info] <0.119.0>@id_generator:init:30 ########### id_generator
2017-05-17 21:14:33.324 [info] <0.120.0>@group_sup:init:82 ######## group_sup
2017-05-17 21:14:33.325 [info] <0.121.0>@client_sup:init:69 ######## client_sup
2017-05-17 21:14:33.325 [error] <0.121.0> CRASH REPORT Process <0.121.0> with 0 neighbours exited with reason: {bad_start_spec,[]} in gen_server:init_it/6 line 344
2017-05-17 21:14:33.325 [error] <0.99.0> Supervisor control_sup had child client_sup started with client_sup:start_link() at undefined exit with reason {bad_start_spec,[]} in context start_error
2017-05-17 21:14:33.326 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}} in context start_error
2017-05-17 21:17:03.465 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:17:03.465 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:17:03.558 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:17:03.565 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:17:03.565 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:17:03.582 [info] <0.119.0>@id_generator:init:30 ########### id_generator
2017-05-17 21:17:03.595 [info] <0.120.0>@group_sup:init:82 ######## group_sup
2017-05-17 21:17:03.596 [info] <0.121.0>@client_sup:init:69 ######## client_sup
2017-05-17 21:17:03.596 [error] <0.121.0> CRASH REPORT Process <0.121.0> with 0 neighbours exited with reason: {bad_start_spec,[]} in gen_server:init_it/6 line 344
2017-05-17 21:17:03.597 [error] <0.99.0> Supervisor control_sup had child client_sup started with client_sup:start_link() at undefined exit with reason {bad_start_spec,[]} in context start_error
2017-05-17 21:17:03.597 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}} in context start_error
2017-05-17 21:17:03.597 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,client_sup,{bad_start_spec,[]}}}}},{chat_app,start,[normal,[]]}} in application_master:init/4 line 134
2017-05-17 21:18:53.857 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:18:53.857 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:18:53.952 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:18:53.959 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:18:53.960 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:18:53.974 [info] <0.119.0>@id_generator:init:30 ########### id_generator
2017-05-17 21:18:53.988 [info] <0.120.0>@group_sup:init:82 ######## group_sup
2017-05-17 21:18:53.989 [info] <0.121.0>@client_sup:init:69 ######## client_sup
2017-05-17 21:18:54.007 [info] <0.99.0>@tcp_agent_sup:start_link:35 ######## AAAA
2017-05-17 21:18:54.007 [info] <0.122.0>@tcp_agent_sup:init:63 ######## AAAA
2017-05-17 21:18:54.007 [info] <0.122.0>@tcp_agent_sup:init:69 ######## tcp_agent
2017-05-17 21:18:54.007 [error] <0.122.0> CRASH REPORT Process <0.122.0> with 0 neighbours exited with reason: {bad_start_spec,[]} in gen_server:init_it/6 line 344
2017-05-17 21:18:54.007 [error] <0.99.0> Supervisor control_sup had child tcp_agent_sup started with tcp_agent_sup:start_link() at undefined exit with reason {bad_start_spec,[]} in context start_error
2017-05-17 21:18:54.007 [error] <0.98.0> Supervisor chat_sup had child control_sup started with control_sup:start_link() at undefined exit with reason {shutdown,{failed_to_start_child,tcp_agent_sup,{bad_start_spec,[]}}} in context start_error
2017-05-17 21:18:54.007 [error] <0.96.0> CRASH REPORT Process <0.96.0> with 0 neighbours exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,tcp_agent_sup,{bad_start_spec,[]}}}}},{chat_app,start,[normal,[]]}} in application_master:init/4 line 134
2017-05-17 21:18:54.008 [info] <0.31.0> Application chat exited with reason: {{shutdown,{failed_to_start_child,control_sup,{shutdown,{failed_to_start_child,tcp_agent_sup,{bad_start_spec,[]}}}}},{chat_app,start,[normal,[]]}}
2017-05-17 21:21:40.108 [info] <0.31.0> Application lager started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:21:40.108 [info] <0.98.0>@chat_sup:init:26 ###############   Begin  ##############
2017-05-17 21:21:40.302 [info] <0.31.0> Application mnesia started on node 'yangwei_node_01@brucedeiMac.local'
2017-05-17 21:21:40.308 [info] <0.99.0>@mod_control_sup:init:37 error
2017-05-17 21:21:40.309 [info] <0.99.0>@control_sup:init:61 
2017-05-17 21:21:40.335 [info] <0.119.0>@id_generator:init:30 ########### id_generator
2017-05-17 21:21:40.336 [info] <0.120.0>@group_sup:init:82 ######## group_sup
2017-05-17 21:21:40.344 [info] <0.121.0>@client_sup:init:69 ######## client_sup
2017-05-17 21:21:40.345 [info] <0.99.0>@tcp_agent_sup:start_link:35 ######## AAAA
2017-05-17 21:21:40.345 [info] <0.122.0>@tcp_agent_sup:init:63 ######## AAAA
2017-05-17 21:21:40.345 [info] <0.122.0>@tcp_agent_sup:init:69 ######## tcp_agent
2017-05-17 21:21:40.347 [info] <0.99.0>@tcp_acceptor:start:17 ######## tcp_acceptor
